src/pudl/extract/ferc1.py

```diff
@@ -897,7 +922,94 @@ def extract_dbf_generic(
+            Contains the tables and years to be loaded into PUDL.
+
+    There are some instances where multiple xbrl tables ought to be combined into
+    a single table to best mesh with data from the other source. This function
+    concatinates those tables into one. It is similar to the extract_dbf_concat
+    except that this function handles the instant and duration tables from xbrl.
+
+    It does not combine instant and duration tables, rather, it creates an instant table
+    that is a combination of several other instant tables, and a duration table that is
+    the combination of several other instant tables (those listed in table_names).
+    """
+    tables = []
+    for raw_table_name in table_names:
+        tables.append(
+            extract_xbrl_generic(
+                table_name=raw_table_name,
+                period=period,
+                ferc1_engine=ferc1_engine,
+                ferc1_settings=ferc1_settings,
+            )
+        )
+
+    combined_table = pd.concat(tables)
+    if combined_table.empty:
+        combined_table = pd.DataFrame()
+    if not len(combined_table) == sum([len(x) for x in tables]):
+        raise AssertionError(
+            "Something went wrong with the concatination and the table lengths don't "
+            "match."
+        )
+    return combined_table
+
+
+def extract_dbf_concat(
+    table_names: list[str],
+    ferc1_settings: Ferc1Settings,
+    pudl_settings: dict[str, Any],
+) -> pd.DataFrame:
+    """Combine multiple raw dbf tables into one.
+
+    Args:
+        table_names: The name of the raw dbf tables you want to combine
+            under xbrl. These are the tables you want to combine.
+        ferc1_settings: Object containing validated settings relevant to FERC Form 1.
+            Contains the tables and years to be loaded into PUDL.
+        pudl_settings: A PUDL settings dictionary.
+        period: Either duration or instant.
+
+    There are some instances where multiple dbf tables ought to be combined into
+    a single table to best mesh with data from the other source. This function
+    concatinates those tables into one. It is similar to the extract_xbr_concat
+    except that this function doesn't have to deal with instant and duration tables.
+    """
+    tables = []
+    for raw_table_name in table_names:
+        tables.append(
+            extract_dbf_generic(
+                ferc1_engine=sa.create_engine(pudl_settings["ferc1_db"]),
+                ferc1_settings=ferc1_settings,
+                table_name=raw_table_name,
+            )
+        )
+    combined_table = pd.concat(tables)
+    if combined_table.empty:
+        combined_table = pd.DataFrame()
+    if not len(combined_table) == sum([len(x) for x in tables]):
+        raise AssertionError(
+            "Something went wrong with the concatination and the table lengths don't "
+            "match."
+        )
+    return combined_table
 
 
 def extract_xbrl_metadata(
```

Comment:
One reason the CI tests are failing is because this newly updated metadata extraction function is now "too complex" meaning there are too many conditionals / loops. Lmk if you have suggestions for simplifying it.

Reply:
we could generate the list of full table names upfront and loop through that. (this suggestion would remove the two assertions that could be raised early on... but idk I don't think they provide a ton of extra context to what the default failure would be)

```
xbrl_table_stems = [
    raw_tbls["xbrl"]
    for (pudl_tbl, raw_tbls) in TABLE_NAME_MAP_FERC1.items()
    if pudl_tbl in ferc1_settings.tables
]
# remove sub-lists of lists if they exist
xbrl_table_stems = pudl.helpers.flatten_list(xbrl_table_stems)
xbrl_tables = [tbl_xbrl + "_instant" for tbl_xbrl in xbrl_table_stems] + [
    tbl_xbrl + "_duration" for tbl_xbrl in xbrl_table_stems
]

Reply:
I'm pretty sure this gets the same results as your code:
```diff
@@ -1042,34 +1056,24 @@ def extract_xbrl_metadata(
     with open(pudl_settings["ferc1_xbrl_taxonomy_metadata"]) as f:
         xbrl_meta_all = json.load(f)

-    xbrl_meta_out = {}
-    for pudl_table in ferc1_settings.tables:
-        if pudl_table not in TABLE_NAME_MAP_FERC1:
-            raise ValueError(f"{pudl_table} not found in the list of known tables.")
-        if "xbrl" not in TABLE_NAME_MAP_FERC1[pudl_table]:
-            raise ValueError(f"No XBRL tables have been associated with {pudl_table}.")
+    valid_tables = {
+        table_name: xbrl_table
+        for table_name in ferc1_settings.tables
+        if (xbrl_table := TABLE_NAME_MAP_FERC1.get(table_name, {}).get("xbrl"))
+        is not None
+    }
+
+    def squash_period(xbrl_table: str | list[str], period, xbrl_meta_all):
+        if type(xbrl_table) is str:
+            xbrl_table = [xbrl_table]
+        return [xbrl_meta_all.get(f"{table}_{period}", []) for table in xbrl_table]
+
+    xbrl_meta_out = {
+        table_name: {
+            "instant": squash_period(xbrl_table, "instant", xbrl_meta_all),
+            "duration": squash_period(xbrl_table, "duration", xbrl_meta_all),
+        }
+        for table_name, xbrl_table in valid_tables.items()
+    }

-        logger.info(
-            f"Reading XBRL Taxonomy metadata for FERC Form 1 table {pudl_table}"
-        )
-        # Attempt to extract both duration and instant tables
-        xbrl_table = TABLE_NAME_MAP_FERC1[pudl_table]["xbrl"]
-        xbrl_meta_out[pudl_table] = {}
-
-        for period in ["instant", "duration"]:
-            # If you're squashing more than one instant and duration table
-            # No need to worry about duplicates b/c they are handled in
-            # transform.ferc1.process_xbrl_metadata(). Here we treat all xbrl_tables
-            # list a list (even if there is just one table in them) to reduce the
-            # amount of code.
-            if type(xbrl_table) is str:
-                xbrl_table = [xbrl_table]
-            period_meta = []
-            for table in xbrl_table:
-                try:
-                    table_meta = xbrl_meta_all[f"{table}_{period}"]
-                except KeyError:
-                    table_meta = []
-                period_meta += table_meta
-            xbrl_meta_out[pudl_table][period] = period_meta
```

Reply:
If you were worried about there being invalid tables at all, you could assert that the length of `valid_tables` is the same as that of `ferc1_settings.tables`.

Reply:
@jdangerx your suggested code for extract_xbrl_metadata  from above was great, except I needed to alter this one piece of code so that it would output nothing `[]` rather than an empty list `[[], [], []]` if there was no metadata. 

I turned this: 
`return [xbrl_meta_all.get(f"{table}_{period}", []) for table in xbrl_table]`

Into this:
```
return [
    metadata
    for table in xbrl_table
    for metadata in xbrl_meta_all.get(f"{table}_{period}", [])
    if metadata
]
```

Reply:
Now it passes the ~complexity requirements~